app/lib/rocketchat.js

```diff
@@ -1003,20 +1004,39 @@ const RocketChat = {
 		return this.sdk.subscribe(...args);
 	},
 	subscribeRoom(...args) {
-		return this.sdk.subscribeRoom(...args);
+		const { server } = reduxStore.getState();
+		const { version: serverVersion } = server;
+		const topic = 'stream-notify-room';
+		let eventUserTyping;
+		if (compareServerVersion(serverVersion, '4.0.0', methods.greaterThanOrEqualTo)) {
+			eventUserTyping = this.subscribe(topic, `${args[0]}/user-activity`, ...args);
+		} else {
+			eventUserTyping = this.subscribe(topic, `${args[0]}/typing`, ...args);
+		}
+
+		// From RocketChat/Rocket.Chat.js.SDK#mobile @lib/drivers/ddp.ts 561 ~ 568
+		return Promise.all([
+			this.subscribe('stream-room-messages', args[0], ...args),
+			eventUserTyping,
+			this.subscribe(topic, `${args[0]}/deleteMessage`, ...args)
+		]);
 	},
 	unsubscribe(subscription) {
 		return this.sdk.unsubscribe(subscription);
 	},
 	onStreamData(...args) {
 		return this.sdk.onStreamData(...args);
 	},
-	emitTyping(room, typing = true) {
-		const { login, settings } = reduxStore.getState();
+	emitUserActivity({ room, activities, extras = {}, activity, performing }) {
```

Comment:
We should have only one param `activities` instead of both `activities` and `activity`.
For the test on older server we can do a `includes` in the array if needed.